src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1659,51 +1659,80 @@ protected override void InvokeDefaultAction(string path)
+                            var fullName = Path.Combine(directory.FullName, filesystemInfo.Name);
+                            var newFilename = InternalTestHooks.GciEnumerationActionRename;
+                            if (String.IsNullOrEmpty(newFilename))
+                            {
+                                File.Delete(fullName);
+                            }
+                            else
+                            {
+                                var newFullName = Path.Combine(directory.FullName, newFilename);
+                                File.Move(fullName, newFullName);
+                            }
                         }
 
-                        bool hidden = false;
-                        if (!Force) hidden = (filesystemInfo.Attributes & FileAttributes.Hidden) != 0;
-
-                        // if "Hidden" is explicitly specified anywhere in the attribute filter, then override
-                        // default hidden attribute filter.
-                        // if specification is to return all containers, then do not do attribute filter on
-                        // the containers.
-                        bool attributeSatisfy =
-                            ((attributeFilter && switchAttributeFilter) ||
-                                ((returnContainers == ReturnContainers.ReturnAllContainers) &&
-                                ((filesystemInfo.Attributes & FileAttributes.Directory) != 0)));
-
-                        if (attributeSatisfy && (filterHidden || switchFilterHidden || Force || !hidden))
+                        try
                         {
-                            if (nameOnly)
+                            bool attributeFilter = true;
+                            bool switchAttributeFilter = true;
+                            bool filterHidden = false;           // "Hidden" is specified somewhere in the expression
+                            bool switchFilterHidden = false;     // "Hidden" is specified somewhere in the parameters
+
+                            if (null != evaluator)
                             {
-                                WriteItemObject(
-                                    filesystemInfo.Name,
-                                    filesystemInfo.FullName,
-                                    false);
+                                attributeFilter = evaluator.Evaluate(filesystemInfo.Attributes);  // expressions
+                                filterHidden = evaluator.ExistsInExpression(FileAttributes.Hidden);
                             }
-                            else
+                            if (null != switchEvaluator)
                             {
-                                if (filesystemInfo is FileInfo)
-                                    WriteItemObject(filesystemInfo, filesystemInfo.FullName, false);
+                                switchAttributeFilter = switchEvaluator.Evaluate(filesystemInfo.Attributes);  // switch parameters
```

Comment:
Please remove the comment or put on separate line before the code and expand it.